Add examples for kubesdk usage

VNadygin opened the pull request:
## Summary

Comprehensive examples demonstrating kubesdk features:

| # | Example | Description |
|---|---------|-------------|
| 01 | quickstart.py | Basic CRUD operations with ConfigMaps |
| 02 | authentication.py | Login patterns: default, specific context, custom path |
| 03 | request_config.py | Timeouts, retries, and logging configuration |
| 04 | deployments.py | Deployment management: create, scale, rolling updates |
| 05 | watching.py | Real-time resource streaming with watch |
| 06 | patching.py | Strategic merge patch, JSON patch, selective field updates |
| 07 | multi_cluster.py | Working with multiple clusters via server= parameter |
| 08 | error_handling.py | Typed exceptions, return_api_exceptions pattern |
| 09 | batch_operations.py | Parallel operations with asyncio.gather |
| 10 | custom_resources.py | CRD example with CloudNativePG Cluster |

Also includes:
- `examples/README.md` with key patterns and usage guide
- Migration guide for v0.1.0 compatibility

## Test plan

- [x] All 10 examples tested against minikube cluster
- [x] Verified resources are created/updated/deleted in Kubernetes
- [x] Verified compatibility with upcoming v0.1.0 (PR #18)

VNadygin committed:
Merge branch 'main' of https://github.com/puzl-cloud/kubesdk

VNadygin committed:
add quickstart example demonstrating basic CRUD operations

Demonstrates:
- Creating, reading, updating, and deleting ConfigMaps
- Listing resources with filtering
- Using create_or_update_k8s_resource for upsert operations
- Proper async/await patterns with kubesdk

VNadygin committed:
add authentication example demonstrating login patterns

VNadygin committed:
add deployments example with CRUD, scaling, and rolling updates

VNadygin committed:
add watch example demonstrating real-time resource streaming

VNadygin committed:
remove demonstrate_automatic_reauth from authentication example

VNadygin committed:
simplify deployments example: remove polling, use return_api_exceptions

VNadygin committed:
add migration guide for v0.1.0 compatibility

Documents required changes to examples when upgrading to kubesdk v0.1.0:
- path_picker module removed, exports moved to root
- new replace_() function for deep nested updates
- updated import patterns for consistency

nik-paushkin commented on examples/02_authentication.py:
```diff
@@ -0,0 +1,134 @@
+    return server_info
+
+
+async def login_with_specific_context():
```
Proper typing is required for each function to make examples clear

VNadygin replied:
Done

VNadygin committed:
add type hints, remove unused imports, clean up examples

- add proper return type hints to all functions
- remove unnecessary namespace fetch in authentication example (login checks API access)
- remove unused imports: NotFoundError, DeleteOptions

VNadygin committed:
add request configuration example, renumber examples

- add 03_request_config.py: timeouts, retries, logging configuration
- rename 03_deployments.py -> 04_deployments.py
- rename 04_watching.py -> 05_watching.py

New order:
  01_quickstart - CRUD basics
  02_authentication - login patterns
  03_request_config - client configuration
  04_deployments - complex resources + patching
  05_watching - event streaming

VNadygin committed:
add patching example demonstrating update strategies

Covers:
- strategic merge patch with built_from_latest
- selective field updates with path picker
- nested field updates
- full replacement (PUT) with force=True
- JSON patch with conflict guards
- conflict handling with retry pattern

VNadygin committed:
add multi-cluster example demonstrating cross-cluster operations

Covers:
- connecting to clusters via kubeconfig contexts
- server= parameter for targeting specific clusters
- parallel queries across clusters

VNadygin committed:
add error handling example demonstrating exception patterns

Shows: typed exceptions, return_api_exceptions, Status object parsing,
batch error aggregation, and login error handling

VNadygin committed:
add batch operations example with parallel create/update/delete

Demonstrates asyncio.gather for parallel operations and label selectors
for batch queries

VNadygin committed:
add custom resources example using CloudNativePG Cluster CRD

Shows how to define K8sResource class for CRDs and perform CRUD operations

VNadygin committed:
update examples README with current file list and patterns

VNadygin committed:
add comprehensive GUIDE.md covering all kubesdk features

nik-paushkin commented on examples/08_error_handling.py:
this is basic Python, isn't needed as a separate example

nik-paushkin commented on examples/10_custom_resources.py:
```diff
@@ -0,0 +1,102 @@
+    patch_strategies_: ClassVar[set] = {PatchRequestType.merge}
+
+    metadata: ObjectMeta = field(default_factory=ObjectMeta)
+    spec: dict | None = None
```
this isn't a valid spec, but I'm gonna write a separate md file on extending kubesdk with CRDs anyway

VNadygin committed:
remove examples 08, 09, 10 - not needed for now

nik-paushkin commented on examples/02_authentication.py:
```diff
@@ -0,0 +1,129 @@
+    # Check connection details
+    await check_connection_info()
+
+    # Uncomment to test other patterns:
```
should be no "uncomment something" sections in examples

VNadygin committed:
Merge remote-tracking branch 'origin/main' into examples/quickstart

VNadygin committed:
use consolidated package imports in authentication example

VNadygin committed:
use consolidated package imports in all examples

VNadygin committed:
remove redundant check_connection_info from auth example

VNadygin committed:
enable all login examples in auth example
